aws/provider_test.go

```diff
@@ -292,6 +292,18 @@ func testAccCheckResourceAttrHostnameWithPort(resourceName, attributeName, servi
 	}
 }
 
+// testAccMatchResourceAttrAccountID ensures the Terraform state regexp matches an account ID
+func testAccMatchResourceAttrAccountID(resourceName, attributeName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		attributeMatch, err := regexp.Compile(`\d{12}`)
```

Comment:
Nit: We'll want to check the whole string so things like `oops123456789012` doesn't match

```suggestion
		attributeMatch, err := regexp.Compile(`^\d{12}$`)
```

Further we can also use `regexp.MustCompile()` instead since its not accepting "user input" for the regular expression to compile:

```go
func testAccMatchResourceAttrAccountID(resourceName, attributeName string) resource.TestCheckFunc {
	return func(s *terraform.State) error {
		return resource.TestMatchResourceAttr(resourceName, attributeName, regexp.MustCompile(`^\d{12}$`))(s)
	}
}
```